packages/common-ui-web/src/components/TakePictureButton/hooks.ts

```diff
@@ -0,0 +1,50 @@
+import { CSSProperties, useMemo } from 'react';
+
+/**
+ * Props that the Monk Take Picture Button component can take.
+ */
+export interface TakePictureButtonProps {
+  /**
+   * Callback for the button when it's clicked.
+   */
+  onClick?: () => void;
+  /**
+   * Boolean indicating if the button is disabled or not.
+   *
+   * @default false
+   */
+  disabled: boolean;
+  /**
+   * This size includes the center circle + the outer rim, not just the circle at the middle.
+   *
+   * @default 60
+   */
+  size: number;
+}
+
+interface TakePictureButtonStyles {
+  innerLater: CSSProperties;
+  outerLayer: CSSProperties;
+}
+
+const buttonBorderWidth = 3;
+const innerButtonMargin = 3;
+
+export function useTakePictureButtonStyle({
+  size = 60,
+}: Partial<TakePictureButtonProps>): TakePictureButtonStyles {
+  return useMemo(() => {
+    return {
+      outerLayer: {
+        width: size - 2 * buttonBorderWidth,
+        height: size - 2 * buttonBorderWidth,
+        borderWidth: buttonBorderWidth,
+      },
+      innerLater: {
+        width: size - 2 * buttonBorderWidth - 2 * innerButtonMargin,
+        height: size - 2 * buttonBorderWidth - 2 * innerButtonMargin,
+        margin: innerButtonMargin,
+      },
+    };
+  }, [size]);
+}
```

Comment:
You missed one important part of the specifications : the border sizes and such are DYNAMIC : they change based on the size of the button. If you look at the specs, i have specified :

> The diameter of the interior circle should be 84% the one of the full circle
> The width of the outer rim should be equal to the width of the margin between the inner circle and the outer circle

This means that you should have something like this instead :

```typescript
const INNER_BUTTON_SIZE_RATIO = 0.84;

export function useTakePictureButtonStyle({
  size = 60,
}: Partial<TakePictureButtonProps>): TakePictureButtonStyles {
  return useMemo(() => {
    const borderWidth = size * (1 - INNER_BUTTON_SIZE_RATIO) / 4;
    return {
      outerLayer: {
        width: size - 2 * borderWidth,
        height: size - 2 * borderWidth,
        borderWidth,
      },
      innerLayer: {
        width: size * INNER_BUTTON_SIZE_RATIO,
        height: size * INNER_BUTTON_SIZE_RATIO,
        margin: borderWidth,
      },
    };
  }, [size]);
}
```

(You have to test this, i have written it like that but i don't know if it works, but you get the idea!)